server/src/models/Rooms.ts

```diff
@@ -3,16 +3,38 @@ import { Room } from './Room';
 class Rooms {
   private readonly rooms: Room[] = [];
 
-  getRoom(roomId: number): Room {
-    //Do wywalenia
-    const room = this.rooms.find(({ id }) => id === roomId) ?? rooms.createRoom(roomId);
+  getRoom(roomId: number): Room | string {
+    const room = this.rooms.find(({ id }) => id === roomId);
+    if (typeof room === 'undefined') return 'No room';
     return room;
   }
 
-  private createRoom(roomId: number): Room {
-    const room = new Room(roomId);
-    this.rooms.push(room);
-    return room;
+  createRoom() {
+    let uniqueId = this.createRandomId();
+    if (this.isRoomFree()) {
+      while (this.checkId(uniqueId)) {
+        uniqueId = this.createRandomId();
+      }
+      const room = new Room(uniqueId);
+      this.rooms.push(room);
+      return room;
+    } else {
+      return new Error();
+    }
+  }
+
+  isRoomFree(): boolean {
+    if (this.rooms.length === 9000) return false;
+    return true;
+  }
+
+  private createRandomId(): number {
+    return Math.floor(Math.random() * 9000 + 1000);
+  }
+
+  private checkId(roomId: number): boolean {
+    const room = this.rooms.find(({ id }) => id === roomId);
```

Comment:
tak jest nieźle ale można by tu było wykorzystać metode some() i zrobić to w jednej linii